test/e2e/lib/helpers/broker_event_transformation_test_with_source.go

```diff
@@ -164,43 +164,21 @@ func BrokerEventTransformationMetricsTestHelper(client *lib.Client, projectID st
 				lib.E2EDummyRespEventType: 1,
 			},
 		},
-		lib.TriggerMetricWithRespCodeAssertion{
-			TriggerMetricAssertion: lib.TriggerMetricAssertion{
-				ProjectID:       projectID,
-				BrokerName:      brokerName,
-				BrokerNamespace: client.Namespace,
-				StartTime:       start,
-				CountPerTrigger: map[string]int64{
-					trigger.Name:     1,
-					respTrigger.Name: 1,
-				},
-			},
-			ResponseCode: http.StatusAccepted,
-		},
-		// Metric from first two delivery attempts (which would fail).
-		lib.TriggerMetricWithRespCodeAssertion{
-			TriggerMetricAssertion: lib.TriggerMetricAssertion{
-				ProjectID:       projectID,
-				BrokerName:      brokerName,
-				BrokerNamespace: client.Namespace,
-				StartTime:       start,
-				CountPerTrigger: map[string]int64{
-					trigger.Name: 2,
-				},
+		lib.TriggerMetricAssertion{
+			ProjectID:       projectID,
+			BrokerName:      brokerName,
+			BrokerNamespace: client.Namespace,
+			StartTime:       start,
+			CountPerTriggerWithRespCode: map[string]int64{
+				lib.TriggerNameWithRespCode(trigger.Name, http.StatusAccepted):     1,
```

Comment:
nit: please make this a struct instead

Reply:
If this would be too difficult feel free to merge as is.

Reply:
You meant make trigger + resp code a struct?

Reply:
Right, my idea was to make the trigger + code key a struct instead of string.